tests/main/prepare-image-reproducible/task.yaml

```diff
@@ -0,0 +1,147 @@
+    STORE_DIR: $(pwd)/fake-store-blobdir
+    STORE_ADDR: localhost:11028
+
+prepare: |
+    if [ "$TRUST_TEST_KEYS" = "false" ]; then
+        echo "This test needs test keys to be trusted"
+        exit
+    fi
+
+    mkdir -p "$ROOT"
+    chown test:test "$ROOT"
+
+restore: |
+    if [ "$TRUST_TEST_KEYS" = "false" ]; then
+        echo "This test needs test keys to be trusted"
+        exit
+    fi
+    
+    #shellcheck source=tests/lib/store.sh
+    . "$TESTSLIB"/store.sh
+    teardown_fake_store "$STORE_DIR"
+    
+    rm -rf "$ROOT"
+
+execute: |
+    if [ "$TRUST_TEST_KEYS" = "false" ]; then
+        echo "This test needs test keys to be trusted"
+        exit
+    fi
+
+    install_snap_to_fakestore() {
+        local SNAP_NAME="$1"
+        local SNAP_REVISION="$2"
+        local SNAP_PATH
+
+        SNAP_PATH=$("$TESTSTOOLS"/snaps-state pack-local "$SNAP_NAME")
+        make_snap_installable "$STORE_DIR" "$SNAP_PATH"
+        cat > snap-"$SNAP_NAME"-decl.json <<EOF
+    {
+        "type": "snap-declaration",
+        "format": "1",
+        "revision": "1",
+        "snap-name": "$SNAP_NAME",
+        "snap-id": "$SNAP_NAME-id",
+        "plugs": {
+            "snapd-control": {
+            "allow-installation": "true",
+            "allow-auto-connection": "true"
+            }
+        }
+    }
+    EOF
+        cat > snap-"$SNAP_NAME"-rev.json <<EOF
+    {
+        "type": "snap-revision",
+        "snap-id": "$SNAP_NAME-id",
+        "snap-revision": "$SNAP_REVISION"
+    }
+    EOF
+        fakestore new-snap-declaration --dir "${STORE_DIR}" "$SNAP_PATH" --snap-decl-json snap-"$SNAP_NAME"-decl.json
+        fakestore new-snap-revision --dir "${STORE_DIR}" "$SNAP_PATH" --snap-rev-json snap-"$SNAP_NAME"-rev.json
+    }
+    
+    # Ensure the following snaps are available on the host snap to avoid the
+    # store script trying to fetch these after setting up the fake store.
+    echo Installing required snaps we will need
+    snap install core
+    snap install remarshal
+    snap install jq
+
+    echo Download snaps needed for the the model
+    snap download core
+    snap download pc
+    snap download pc-kernel
+    
+    #shellcheck source=tests/lib/store.sh
+    . "$TESTSLIB"/store.sh
+
+    echo "Configure daemon to point to the fake store"
+    setup_fake_store "$STORE_DIR"
+
+    echo Expose the needed assertions through the fakestore
+    cp "$TESTSLIB"/assertions/testrootorg-store.account-key "$STORE_DIR/asserts"
+    cp "$TESTSLIB"/assertions/developer1.account "$STORE_DIR/asserts"
+    cp "$TESTSLIB"/assertions/developer1.account-key "$STORE_DIR/asserts"
+
+    # It is not enough to copy the assertions, we must also ack them otherwise we
+    # will get an error about not being able to resolve the account key
+    snap ack "$STORE_DIR/asserts/testrootorg-store.account-key"
+    snap ack "$STORE_DIR/asserts/developer1.account"
+    snap ack "$STORE_DIR/asserts/developer1.account-key"
```

Comment:
can you tell me where it fails? because this is actually fairly confusing, the acks are not directly relevant for prepare-image, so it's something going on with the tooling for or with the fakestore that is not quite doing the right thing/setup the right way